pointer/pointer.go

```diff
@@ -107,304 +70,218 @@ func Int32Equal(a, b *int32) bool {
 	return *a == *b
 }
 
-// Uint returns a pointer to an uint
-func Uint(i uint) *uint {
-	return &i
-}
+var (
 
-// UintPtr is a function variable referring to Uint.
-//
-// Deprecated: Use Uint instead.
-var UintPtr = Uint // for back-compat
+	// Int returns a pointer to an int
+	Int = Ref[int]
 
-// UintDeref dereferences the uint ptr and returns it if not nil, or else
-// returns def.
-func UintDeref(ptr *uint, def uint) uint {
-	if ptr != nil {
-		return *ptr
-	}
-	return def
-}
+	// IntPtr is a function variable referring to Int.
+	//
+	// Deprecated: Use Int instead.
+	IntPtr = Int // for back-compat
 
-// UintPtrDerefOr is a function variable referring to UintDeref.
-//
-// Deprecated: Use UintDeref instead.
-var UintPtrDerefOr = UintDeref // for back-compat
+	// IntDeref dereferences the int ptr and returns it if not nil, or else
+	// returns def.
+	IntDeref = Deref[int]
 
-// Uint32 returns a pointer to an uint32.
-func Uint32(i uint32) *uint32 {
-	return &i
-}
+	// IntPtrDerefOr is a function variable referring to IntDeref.
+	//
+	// Deprecated: Use IntDeref instead.
```

Comment:
Same here: this might as well point to `Deref` directly. (Same goes for all the other instances.)